libs/balances-and-allowances/src/hooks/usePersistBalancesFromBff.ts

```diff
@@ -114,18 +152,21 @@ export async function getBffBalances(
 
   try {
     const res = await fetch(fullUrl)
-    const data: BalanceResponse = await res.json()
+    const data = await parseBffResponse(res)
 
     if (!res.ok) {
-      return Promise.reject(new Error(`BFF error: ${res.status} ${res.statusText}`))
+      handleBffError(res, data)
     }
 
-    if (!data.balances) {
+    if (!('balances' in data) || !data.balances) {
       return null
     }
 
     return data.balances
   } catch (error) {
-    return Promise.reject(error)
+    if (error instanceof Error && isUnsupportedChainError(error.message)) {
+      throw new Error('Unsupported chain')
```

Comment:
Let's more `'Unsupported chain'` in a const, or make a custom error for the case:
```
export class UnsupportedChain extends Error {...}

function isUnsupportedChainError(error: Error): error is UnsupportedChain {
  return error instanceof UnsupportedChain
}
```

Reply:
done